tests/compass-runtime-agent/test/runtimeagent/test/runtime_agent_test.go

```diff
@@ -0,0 +1,410 @@
+				application := this.initialPhaseResult
+
+				// updating whole application
+				updatedInput := applications.NewApplication("test-app-2-updated", "", map[string][]string{}).
+					WithAPIs(
+						[]*applications.APIDefinitionInput{
+							noAuthAPIInput,
+							basicAuthAPIInput,
+							oauthAPIInput,
+						}).
+					WithEventAPIs(
+						[]*applications.EventAPIDefinitionInput{
+							applications.NewEventAPI("events-api", "description").WithJsonEventApiSpec(&apiEventSpecDataCreate),
+						},
+					)
+
+				updatedApp, err := testSuite.CompassClient.UpdateApplication(application.ID, updatedInput.ToCompassInput())
+				require.NoError(t, err)
+				assert.Equal(t, 3, len(updatedApp.APIs.Data))
+
+				apiIds := getAPIsIds(updatedApp)
+				t.Logf("Updated APIs for %s Application", updatedApp.ID)
+				logIds(t, apiIds)
+				t.Logf("Adding denier labels for %s Application", updatedApp.ID)
+				testSuite.AddDenierLabels(t, updatedApp.ID, apiIds...)
+
+				// then
+				this.secondPhaseAssert = func(t *testing.T, testSuite *runtimeagent.TestSuite, this *testCase) {
+					// assert previous APIs deleted
+					for _, api := range application.APIs.Data {
+						testSuite.K8sResourceChecker.AssertAPIResourcesDeleted(t, application.ID, api.ID)
+					}
+					// assert previous EventAPIs deleted
+					for _, eventAPI := range application.EventAPIs.Data {
+						testSuite.K8sResourceChecker.AssertAPIResourcesDeleted(t, application.ID, eventAPI.ID)
+					}
+
+					// assert updated Application
+					testSuite.K8sResourceChecker.AssertResourcesForApp(t, updatedApp)
+					testSuite.APIAccessChecker.AssertAPIAccess(t, updatedApp.APIs.Data...)
+				}
+			},
+		},
+		{
+			description: "Test case 3: Change auth in all APIs",
+			initialPhaseInput: func() *applications.ApplicationInput {
+				return applications.NewApplication("test-app-3", "", map[string][]string{}).
+					WithAPIs(
+						[]*applications.APIDefinitionInput{
+							applications.NewAPI("no-auth-api", "no auth api", testSuite.GetMockServiceURL()),
+							applications.NewAPI("basic-auth-api", "basic auth api", testSuite.GetMockServiceURL()).WithAuth(basicAuth),
+							applications.NewAPI("oauth-auth-api", "oauth api", testSuite.GetMockServiceURL()).WithAuth(oauth),
+						}).
+					WithEventAPIs(
+						[]*applications.EventAPIDefinitionInput{
+							applications.NewEventAPI("events-api", "description"),
+							applications.NewEventAPI("no-description-events-api", ""),
+						},
+					)
+			},
+			initialPhaseAssert: assertK8sResourcesAndAPIAccess,
+			secondPhaseSetup: func(t *testing.T, testSuite *runtimeagent.TestSuite, this *testCase) {
+				// when
+				application := this.initialPhaseResult
+
+				var updatedAPIs []*graphql.APIDefinition
+
+				// update no auth API to OAuth
+				noAuthAPI, found := getAPIByName(application.APIs.Data, "no-auth-api")
+				require.True(t, found)
+				updatedInput := applications.NewAPI("basic-to-oauth", "", noAuthAPI.TargetURL).WithAuth(oauth)
```

Comment:
"no-auth-to-oauth" instead of "basic-to-oauth"?

Reply:
done